bricks/distributed/test.cc

```diff
@@ -0,0 +1,145 @@
+
+  // Merge incorrect update after previous update
+  c2 = {1, 2};
+  // can't merge T > T'
+  EXPECT_EQ(false, v.merge(c2));
+  // invalid data, merged vector
+  EXPECT_EQ(v.state()[0], cur_state[0] + 1);
+  EXPECT_EQ(v.state()[1], cur_state[1]);
+
+  // Merge partially equals using lte validation
+  v = VectorClock(c1, 0);
+  c2 = {1, 3};
+  // 0 is equeal, 1 is greater - ok to merge
+  EXPECT_EQ(true, v.merge(c2));
+  cur_state = v.state();
+  // local time should be updated after merge
+  EXPECT_GT(cur_state[0], c2[0]);
+  // merged time should be equal c2[1]
+  EXPECT_EQ(c2[1], cur_state[1]);
+
+  // Merge partially incorrect
+  v = VectorClock(c1, 0);
+  cur_state = v.state();
+  c2 = DiscreteClocks({1, 0});
+  EXPECT_EQ(false, v.merge(c2));
+  // invalid data, merged vector
+  EXPECT_EQ(v.state()[0], cur_state[0] + 1);
+  EXPECT_EQ(v.state()[1], cur_state[1]);
+}
+
+TEST(VectorClock, ContinuousTime) {
+  auto base_time = current::time::Now();
+  ContinuousClocks c1 = {base_time, base_time + std::chrono::microseconds(100)};
+  auto v = VectorClock<ContinuousClocks, MergeStrategy>(c1, 0);
+
+  // Merge correct update
+  ContinuousClocks c2 = {base_time + std::chrono::microseconds(200), base_time + std::chrono::microseconds(300)};
+  // each element is greater - ok to merge
+  EXPECT_EQ(true, v.merge(c2));
+  auto cur_state = v.state();
+
+  // Merge incorrect update after previous update
+  c2 = {base_time + std::chrono::microseconds(100), base_time + std::chrono::microseconds(200)};
+  // can't merge T > T'
+  EXPECT_EQ(false, v.merge(c2));
+}
+
+TEST(VectorClock, StrictMerge) {
+  auto base_time = current::time::Now();
+  DiscreteClocks c1 = {1, 2};
+  auto v = VectorClock<DiscreteClocks, StrictMergeStrategy>(c1, 0);
+
+  // Merge correct update
+  DiscreteClocks c2 = {2, 3};
+  // each element is greater - ok to merge
+  EXPECT_EQ(true, v.merge(c2));
+  auto cur_state = v.state();
+  // local time should be updated after merge
+  EXPECT_GT(cur_state[0], c2[0]);
+  // merged time should be equal c2[1]
+  EXPECT_EQ(c2[1], cur_state[1]);
+
+  // Merge equals using strict validation
+  c1 = {10, 20};
+  v = VectorClock<DiscreteClocks, StrictMergeStrategy>(c1, 0);
+  cur_state = v.state();
+  c2 = {10, 20};
+  EXPECT_EQ(false, v.merge(c2));
+  EXPECT_EQ(v.state()[0], cur_state[0] + 1);
+  EXPECT_EQ(v.state()[1], cur_state[1]);
+
+  // Merge partially equals
+  v = VectorClock<DiscreteClocks, StrictMergeStrategy>(c1, 0);
+  cur_state = v.state();
+  c2 = {1, 20};
+  // 0 is equeal, 1 is greater - not ok to merge
+  EXPECT_EQ(false, v.merge(c2));
+  EXPECT_EQ(v.state()[0], cur_state[0] + 1);
+  EXPECT_EQ(v.state()[1], cur_state[1]);
+
+  // Merge incorrect
+  v = VectorClock<DiscreteClocks, StrictMergeStrategy>(c1, 0);
+  cur_state = v.state();
+  c2 = {0, 1};
+  EXPECT_EQ(false, v.merge(c2));
+  // internal state was not changed
+  EXPECT_EQ(v.state()[0], cur_state[0] + 1);
+  EXPECT_EQ(v.state()[1], cur_state[1]);
+}
```

Comment:
Nit: No newline at end of file.

(Did you try `make indent` / `make fmt`, with `clang-format`?)